### DISCUSSION TITLE
Sharing data between getStaticProps and getStaticPaths

### DISCUSSION THREAD

**Discussion by pomber** [vercel/next.js#11370] [Help]
Hi!

I need to fetch some data that I use in `getStaticProps` and in `getStaticPaths`:

```js
const url = "http://some.json";

export async function getStaticProps(context) {
  const { name } = context.params;
  const data = await fetch(url).then(r => r.json());
  const foo = data[name];
  return { props: { name, foo } };
}

export async function getStaticPaths() {
  const data = await fetch(url).then(r => r.json());
  return {
    paths: Object.keys(data).map(name => ({ params: { name } })),
    fallback: false
  };
}
```

Note that I'm fetching `"http://some.json"` in both functions.

**What's an idiomatic way to share the data between the two functions (and between each call to `getStaticProps`)?**

Is this OK? Will the fetch also happen client-side?
```js
const url = "http://some.json";
const promise = fetch(url).then(r => r.json());

export async function getStaticProps(context) {
  const { name } = context.params;
  const data = await promise;
  const foo = data[name];
  return { props: { name, foo } };
}

export async function getStaticPaths() {
  const data = await promise;
  return {
    paths: Object.keys(data).map(name => ({ params: { name } })),
    fallback: false
  };
}
```

**Comment by ijjk** [comment]
Hi, you might want to take a look at [my reply here](https://github.com/zeit/next.js/issues/10933#issuecomment-598297975) which describes some ways you can achieve caching data re-used between these methods currently. A discussion has also been started at https://github.com/zeit/next.js/discussions/11272 for the same topic.

> Is this OK? Will the fetch also happen client-side?

Note: `getStaticPaths` and `getStaticProps` is only called on the server at build time and on the client only requests the build time generated data file

**Comment by jrandeniya** [comment]
I've implemented the file-based cache solution here if anyone is interested. I needed it for my own project:
[jrandeniya/nextjs-ssg-contentful-cached](https://github.com/jrandeniya/nextjs-ssg-contentful-cached).

It's in Typescript with build-time type generation and caching calls between `getStaticProps` and `getStaticPaths`.
